### PR TITLE
Trans proteomic pipeline

### PR DESCRIPTION
**Pull request by iracooke**
This formula is for the Trans Proteomic Pipeline, a large and well established collection of proteomics tools.

### CODE REVIEW

**Comment by iracooke** [comment]
Any advice on fixing this would be much appreciated. 
It builds fine on OSX but fails due exceeding the github API rate limit.

**Comment by dpo** [comment]
I restarted the TravisCI build. It's normally a temporary issue.

File: trans_proteomic_pipeline.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+class TransProteomicPipeline < Formula
+  desc "TransProteomicPipeline: Collection of proteomics tools"
+  homepage "http://tools.proteomecenter.org/wiki/index.php?title=Software:TPP"
+  url "https://github.com/iracooke/tpp/archive/v4.8.0p7.tar.gz"
+  version "4.8.0"
+  sha256 "ca335910e65d0c4ba26d5a09de5c541d60261f82a67cf0c6b463c67d6ddad08e"
+
+  depends_on "gd" => :build
+  depends_on "homebrew/versions/libpng12" => :build
+  depends_on "gnuplot"
```
Comment (dpo) [inline trans_proteomic_pipeline.rb]: `gnuplot` already depends on `libpng`, so you're getting a conflict.
@sjackman Does Linuxbrew not honor `keg_only`?

File: trans_proteomic_pipeline.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
+class TransProteomicPipeline < Formula
+  desc "TransProteomicPipeline: Collection of proteomics tools"
+  homepage "http://tools.proteomecenter.org/wiki/index.php?title=Software:TPP"
+  url "https://github.com/iracooke/tpp/archive/v4.8.0p7.tar.gz"
+  version "4.8.0"
+  sha256 "ca335910e65d0c4ba26d5a09de5c541d60261f82a67cf0c6b463c67d6ddad08e"
```
Comment (dpo) [inline trans_proteomic_pipeline.rb]: It would be good to also provide the `head`.

Comment (iracooke) [inline trans_proteomic_pipeline.rb]: Thanks. I just added that

File: trans_proteomic_pipeline.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      ENV.delete("CC")
+      ENV.delete("CXX")
+
+      system "make"
+      system "make", "install"
+    end
+  end
+
+  test do
+    system "#{bin}/ProteinProphet", "-h"
```
Comment (iracooke) [inline trans_proteomic_pipeline.rb]: I totally agree that it would be good if this toolsuite had proper tests.  I'm a bit reluctant to head down that road though.  The original developers of the tools haven't built a proper test suite and I feel like the amount of time I can commit to this is just to have a usable OSX build.  It might be possible to add a more meaningful test for one or two tools but to be honest I'm not sure that would be much more meaningful than a simple (does the binary run .. test).  A proper test suite is just way more work than I can commit too I'm afraid.  I hope we can still merge without this.

**Comment by jonchang** [comment]
@BrewTestBot test this please

**Comment by jonchang** [comment]
@BrewTestBot test this please

File: trans_proteomic_pipeline.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      end
+
+      mkdir_p prefix/"web"
+      mkdir_p prefix/"cgi-bin"
+      mkdir_p prefix/"params"
+
+      ENV.deparallelize
+
+      ENV.delete("CC")
+      ENV.delete("CXX")
```
Comment (sjackman) [inline trans_proteomic_pipeline.rb]: Why are these necessary? Please include a comment in the source code.

**Comment by iracooke** [comment]
@sjackman I'm getting "GitHub repository not notable enough (<20 forks, <20 watchers and <50 stars)" which is totally true.   Is there a way around this automatic check though.  The original source project is very notable but is not on github.  I created a github repo to track the changes I needed to make to the original in order to make it build.

**Comment by sjackman** [comment]
Yes, we can ignore this audit failure. For Homebrew-science, being published is sufficient.

File: trans_proteomic_pipeline.rb [hunk 5]
```diff
@@ -0,0 +1,39 @@
+class TransProteomicPipeline < Formula
+  desc "TransProteomicPipeline: Collection of proteomics tools"
+  homepage "http://tools.proteomecenter.org/wiki/index.php?title=Software:TPP"
+  url "https://github.com/iracooke/tpp/archive/v4.8.0p9.tar.gz"
+  version "4.8.0"
+  sha256 "2f500c867e86eee16e8be1c91a97d1c4e51cfba65a1a3b3fa2327e2731f28d80"
```
Comment (sjackman) [inline trans_proteomic_pipeline.rb]: Please add two ruby comments:
```
# doi "10.1007/978-1-60761-444-9_15"
# tag "bioinformatics"
```

File: trans_proteomic_pipeline.rb [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+  depends_on "gd" => :build
+  depends_on "homebrew/versions/libpng12" => :build
```
Comment (sjackman) [inline trans_proteomic_pipeline.rb]: This formula attempts to install both `libpng12` and `libpng`, because both `gd` and `gnuplot` depend on `libpng`, and `libpng12` and `libpng` conflict with each other. Any chance TPP can be built using `libpng`?

Why are `gd` and `libpng12` only `:build` dependencies? Are their libraries not needed at run time?

File: trans_proteomic_pipeline.rb (hunk 5 above)
Comment (iracooke) [inline trans_proteomic_pipeline.rb]: Sorry this took so long.  Done now.

File: trans_proteomic_pipeline.rb (hunk 6 above)
Comment (iracooke) [inline trans_proteomic_pipeline.rb]: Thanks for this suggestion.  I was surprised to discover that building without libpng12 works.  As far as I can see from the automated tests this builds OK but is just failing due to the lack of notability on OSX.  I haven't got this to work with linuxbrew yet which is why the travis test is failing.  Do I need to get this working with linuxbrew to be accepted?

**Comment by jonchang** [comment]
@BrewTestBot test this please

**Comment by jonchang** [comment]
@BrewTestBot test this please

**Comment by jonchang** [comment]
Thanks for contributing to Homebrew @iracooke! This has been merged.
